homeassistant/data_entry_flow.py

```diff
@@ -495,6 +506,9 @@ class FlowHandler:
     VERSION = 1
     MINOR_VERSION = 1
 
+    __progress_task: asyncio.Task | None = None
+    progress_fut: asyncio.Future[None] | None = None
```

Comment:
I think we should avoid exposing futures to implementors of flows. That's a good rule of thumb in general for asyncio. Aim for an async await syntax instead with tasks of needed.

Reply:
That rule makes no sense, because tasks are futures:
![image](https://github.com/home-assistant/core/assets/14281572/cb46ab70-01bc-4411-a011-6f8cf8f819a6)

Reply:
I think using a Task keeps the async await syntax also internally on this side of the task handling which is nice, since we can't set a result or exception.